plugins/inputs/net/net.go

```diff
@@ -129,6 +133,29 @@ func (n *NetIOStats) Gather(acc telegraf.Accumulator) error {
 	return nil
 }
 
+func getInterfaceSpeed(ioName string) uint64 {
+	sysPath := os.Getenv("HOST_SYS")
+	if sysPath == "" {
+		sysPath = "/sys"
+	}
+
+	raw, err := os.ReadFile(filepath.Join(sysPath, "class", "net", ioName, "speed"))
+	if err != nil {
+		return 0
+	}
+	speedStr := string(raw)
+
+	if speedStr == "-1" {
+		return 0
+	}
```

Comment:
If the value is -1 shouldn't we return that rather than hide that from the user? -1 indicates unsupported? I see it for a number of my NICs/bridges:

```s
❯ ls /sys/class/net/*
/sys/class/net/br-66355ad077c6@  /sys/class/net/incusbr0@
/sys/class/net/br-87021ad10ced@  /sys/class/net/lo@
/sys/class/net/docker0@          /sys/class/net/virbr0@
/sys/class/net/enp6s0@
❯ cat /sys/class/net/*/speed
-1
-1
-1
2500
-1
cat: /sys/class/net/lo/speed: Invalid argument
-1
```

That would mean the return value needs to be updated. Can you also add a test case for -1 to ensure we support that in future changes.

Thanks!